### ISSUE TITLE
BUG: Replicability issues when explaining predictions using the Permutation Explainer

### ISSUE THREAD

**Issue by ArturoSbr** [shap/shap#3995]
### Issue Description

I first came across this issue in [this thread](https://github.com/shap/shap/issues/2118#issuecomment-1639214745). My example concerns `PermutationExplainer`, but I think it may be affecting `KernelExplainer` too.

---

I have a binary sklearn `StackingClassifier` instance that I'm trying to explain with `shap.PermutationExplainer`. My team needs to persist these explainers, so I'm interested in using the library's `save` and `load` functionalities.

Upon running a few tests, I noticed that my explainer does not generate the same shap values when explaining the same data. That is, the `values` attribute of `explainer(data)` differs from one execution to another, even when `explainer` was initialized with a seed.

### Minimal Reproducible Example

```python
# Imports
import shap
import numpy as np
from sklearn.datasets import load_breast_cancer
from sklearn.ensemble import StackingClassifier, RandomForestClassifier, GradientBoostingClassifier

# Set seed
SEED = 42
np.random.seed(SEED)

# Load data
X, y = load_breast_cancer(return_X_y=True)

# Declare stacking classifier
stacking = StackingClassifier(
    estimators=[('rf', RandomForestClassifier()), ('gb', GradientBoostingClassifier())]
).fit(X, y)

# Declare permutation explainer
explainer = shap.explainers.Permutation(
    model=stacking.predict_proba,  # Binary classifier
    masker=X,
    seed=SEED
)

# Save explainer
with open('permutation-explainer.pkl', 'wb') as f:
    explainer.save(f)

# Load explainer
with open('permutation-explainer.pkl', 'rb') as f:
    e2 = shap.Explainer.load(f)

# Data to explain
test = X[:100]

# Declare explanations
v1 = explainer(test)  # Original object
v2 = e2(test)  # Loaded object
v3 = explainer(test)  # Original object again, just to check

# v1 VS v2
(v1.values[:, :, 1] - v2.values[:, :, 1] < 1e-03).mean()  # != 1

# v1 VS v3
(v1.values[:, :, 1] - v3.values[:, :, 1] < 1e-03).mean()  # != 1
```

### Traceback

```shell

```

### Expected Behavior

Objects  `v1`, `v2` and `v3` should all be similar to one another (if not identical!).

### Bug report checklist

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [x] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

```
Name: shap
Version: 0.46.0
Summary: A unified approach to explain the output of any machine learning model.
Home-page: 
Author: 
Author-email: Scott Lundberg <[slund1@cs.washington.edu](mailto:slund1@cs.washington.edu)>
License: MIT License
Location: /usr/local/lib/python3.11/dist-packages
Requires: cloudpickle, numba, numpy, packaging, pandas, scikit-learn, scipy, slicer, tqdm
Required-by:
```

**Comment by PredictiveManish** [comment]
<img width="772" height="266" alt="Image" src="https://github.com/user-attachments/assets/0e577c3b-91c6-4129-887b-38a229a7b3cd" />

I have ran the same code and this is the output, @ArturoSbr @CloseChoice is it solved or there's something I'm missing??
